objects/src/block/mod.rs

```diff
@@ -1,6 +1,71 @@
+use alloc::vec::Vec;
+
 use super::{Digest, Felt, Hasher, ZERO};
 
 mod header;
 pub use header::BlockHeader;
 mod note_tree;
 pub use note_tree::BlockNoteTree;
+
+use crate::{
+    notes::Nullifier,
+    transaction::{AccountUpdateData, OutputNote},
+};
+
+pub type NoteBatch = Vec<OutputNote>;
+
+/// A block in the block chain.
+#[derive(Debug, Clone)]
+pub struct Block {
+    /// Block header.
+    header: BlockHeader,
+
+    /// Account updates for the block.
+    updated_accounts: Vec<AccountUpdateData>,
+
+    /// Note batches created in transactions in the block.
+    created_notes: Vec<NoteBatch>,
+
+    /// Nullifiers produced in transactions in the block.
+    produced_nullifiers: Vec<Nullifier>,
+    // TODO:
+    // - full states for created public notes
+    // - zk proof
+}
+
+impl Block {
```

Comment:
May be good to add something like `notes()` method which would look something like this:
```Rust
pub fn notes(&self) -> impl Iterator<Item = (BlockNoteIndex, &OutputNote)>
```

Related to this, but maybe for a different PR: we should probably use `BlockNoteIndex` struct in `BlockNoteTree` methods.